File: scripts/commands/generic/cmdBroadcastPlanet.py
from intents.chat import ChatBroadcastIntent
from intents.chat.ChatBroadcastIntent import BroadcastType

import sys

def execute(objManager, player, target, args):
	ChatBroadcastIntent(args, player, player.getCreatureObject().getLocation().getTerrain(), BroadcastType.PLANET).broadcast()
	return

File: src/main/java/com/projectswg/holocore/resources/support/global/commands/callbacks/chat/CmdBroadcastPlanet.java
package com.projectswg.holocore.resources.support.global.commands.callbacks.chat;

import com.projectswg.holocore.intents.support.global.chat.SystemMessageIntent;
import com.projectswg.holocore.resources.support.global.commands.ICmdCallback;
import com.projectswg.holocore.resources.support.global.player.Player;
import com.projectswg.holocore.resources.support.objects.swg.SWGObject;
import com.projectswg.holocore.resources.support.objects.swg.creature.CreatureObject;

public final class CmdBroadcastPlanet implements ICmdCallback {
	
	@Override
	public void execute(Player player, SWGObject target, String args) {
		CreatureObject creature = player.getCreatureObject();
		if (creature == null)
			return;
		SystemMessageIntent.broadcastPlanet(creature.getTerrain(), args);
	}
	
}
